crates/cubecl-common/src/benchmark.rs

```diff
@@ -7,9 +7,32 @@ use serde::{Deserialize, Serialize};
 
 use super::stub::Duration;
 
+/// How a benchmark's execution times are measured.
+#[derive(Debug, Default, Clone, Copy, Serialize, Deserialize)]
+pub enum TimingMethod {
+    /// Time measurements come from CPU timing of execution + sync
+    /// calls.
+    #[default]
+    CPU,
+    /// Time measurements come from hardware reported timestamps
+    /// coming from a sync call.
+    GPU,
```

Comment:
I think CPU & GPU is a bit confusing, since the execution isn't on the CPU. I would rename those based on the scope, maybe `Full` and `Kernel` or `Total` and `Device`. I'm open to better names 😅. Kernel is kinda bad, since we may benchmark multiple kernels at the same time.

Reply:
Also, we could have two `sync` methods, one that reads the timestamps and returns a duration, and another sync that doesn't read the timestamps. Maybe `sync_profiled() -> Result<Duration, SyncProfiledError>` and `sync() `. At some point we might have the option to start collecting timestamps, and calling `sync_profiled` would return an error, no timestamps recorded. So maybe instead of `sync_profiled` is could be `read_timestamps`, unsure ...

Reply:
> I think CPU & GPU is a bit confusing, since the execution isn't on the CPU. I would rename those based on the scope, maybe `Full` and `Kernel` or `Total` and `Device`. I'm open to better names 😅. Kernel is kinda bad, since we may benchmark multiple kernels at the same time.

Got it, makes sense. How do you feel about DeviceOnly and Full? DeviceOnly implies there is more to benchmark, and Full implies...well, everything. If Full is too vague, maybe DeviceAndOverhead to be more concrete?

Reply:
> Also, we could have two `sync` methods, one that reads the timestamps and returns a duration, and another sync that doesn't read the timestamps. Maybe `sync_profiled() -> Result<Duration, SyncProfiledError>` and `sync() `. At some point we might have the option to start collecting timestamps, and calling `sync_profiled` would return an error, no timestamps recorded. So maybe instead of `sync_profiled` is could be `read_timestamps`, unsure ...

Yeah, having the option to sync without profiling would be nice. It is a little strange that right now sync is forced to also profile, and must incur a profiling cost 😅. Especially since on wgpu it seems that such overhead is non-trivial.